frontend_lib/src/component/Button/IconButton.jsx

```diff
@@ -1,31 +1,55 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-const IconButton = props =>
-  <button
-    className={`btn ${props.className}`}
-    onClick={props.onClick}
-    disabled={props.disabled}
-    style={props.style}
-  >
-    <i className={`fa fa-fw fa-${props.icon}`} /> {props.text}
-  </button>
+import { darkenColor } from '../../helper.js'
+
+require('../../css/IconButton.styl')
+
+const IconButton = props => {
+  const classes = ['iconbutton']
+  classes.push('iconbutton__' + props.mode)
+  classes.push('iconbutton__' + props.intent)
+  classes.push(`iconbutton__${props.intent}_${props.mode}`)
+  if (props.customClass) classes.push(props.customClass)
+  const className = classes.join(' ')
+
+  const style = {
+    '--primaryColor': props.color,
+    '--primaryDarkColor': darkenColor(props.color)
+  }
+  return (
+    <button
+      className={className}
+      style={style}
+      onClick={props.onClick}
+      disabled={props.disabled}
+      data-cy={props.dataCy}
+    >
+      <i className={`fa fa-fw fa-${props.icon}`} /> {props.text}
+    </button>
+  )
+}
 
 IconButton.propTypes = {
   icon: PropTypes.string.isRequired,
-  className: PropTypes.string,
+  text: PropTypes.string.isRequired,
+  color: PropTypes.string,
   onClick: PropTypes.func,
   disabled: PropTypes.bool,
-  text: PropTypes.string,
-  style: PropTypes.object
+  intent: PropTypes.oneOf(['primary', 'secondary']),
+  mode: PropTypes.oneOf(['dark', 'light']),
+  customClass: PropTypes.string,
+  dataCy: PropTypes.string
 }
 
 IconButton.defaultProps = {
-  className: '',
-  onClick: () => {},
+  onClick: undefined,
   disabled: false,
-  text: '',
-  style: {}
+  color: GLOBAL_primaryColor, // eslint-disable-line no-undef
```

Comment:
Wouldn't it be better to GLOBAL_primaryColor in field `globals` of `package.json`?

Reply:
it is set by the backend in index.max
that's why it needs to be a gobal
I agree that it's not ideal but that's the easier solution

Reply:
I suggested to add `GLOBAL_primaryColor` to the `globals` standard's field of package.json to avoid disabling the eslint rule on this line